### PR TITLE
DADA2

### PR DESCRIPTION
**Pull request by TuomasBorman**
Hi

here is the DADA2 workflow. I referred official [DADA2 tutorial](https://benjjneb.github.io/dada2/tutorial.html).

Issue: https://github.com/microbiome/OMA/issues/117

-Tuomas

### CODE REVIEW

File: dada2_workflow.Rmd [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+This example include only the code. The pipelines ends to _TreeSummarizedExperiment_.
+
+```{r dada2_1}
+# GETTING READY
+
+library(dada2); packageVersion("dada2")
+```
+
+```{r dada2_2}
+path <- "~/Desktop/dada2_example/MiSeq_SOP" # CHANGE ME to the directory containing the fastq files after unzipping.
```
Comment (antagomir) [inline dada2_workflow.Rmd:16]: Do not use local paths here. Just something like "data/MiSeq_SOP" or "data/dada2/MiSeq_SOP"

File: dada2_workflow.Rmd [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+colnames(track) <- c("input", "filtered", "denoisedF", "denoisedR", "merged", "nonchim")
+rownames(track) <- sample.names
+head(track)
+```
+
+```{r dada_20}
+# ASSIGN TAXONOMY
+
+# Change path
+taxa <- assignTaxonomy(seqtab.nochim, "~/Desktop/dada2_example/tax/silva_nr_v132_train_set.fa.gz", multithread=TRUE)
```
Comment (antagomir) [inline dada2_workflow.Rmd:129]: no local paths. 

Also cite the DADA2 website that provides now more updated reference databases in the readily usable format (I sent the link through slack earlier, can resend if necessary)

File: dada2_workflow.Rmd [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+```
+
+```{r dada2_21}
+# Change path
+taxa <- addSpecies(taxa, "~/Desktop/dada2_example/tax/silva_species_assignment_v132.fa.gz")
```
Comment (antagomir) [inline dada2_workflow.Rmd:134]: no local paths, check from everywhere

File: dada2_workflow.Rmd [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+cat("DADA2 inferred", length(unqs.mock), "sample sequences present in the Mock community.\n")
+```
+
+```{r dada2_24}
+mock.ref <- getSequences(file.path(path, "HMP_MOCK.v35.fasta"))
+match.ref <- sum(sapply(names(unqs.mock), function(x) any(grepl(x, mock.ref))))
+cat("Of those,", sum(match.ref), "were exact matches to the expected reference sequences.\n")
+```
+
+## Create a TreeSE from the data
```
Comment (antagomir) [inline dada2_workflow.Rmd:157]: remove " a "

Comment (antagomir) [reply dada2_workflow.Rmd:157]: or just: "Create a TreeSE data object"

File: dada2_workflow.Rmd [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+day <- as.integer(sapply(strsplit(samples.out, "D"), `[`, 2))
+samdf <- data.frame(Subject=subject, Gender=gender, Day=day)
+samdf$When <- "Early"
+samdf$When[samdf$Day>100] <- "Late"
+rownames(samdf) <- samples.out
+```
+
+```{r dada2_27}
+counts <- t(seqtab.nochim)
+assays <- SimpleList(counts = counts)
```
Comment (antagomir) [inline dada2_workflow.Rmd:189]: more clear to put this directly on line 194?

File: dada2_workflow.Rmd [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+
+samdf <- DataFrame(samdf)
```
Comment (antagomir) [inline dada2_workflow.Rmd:191]: just on line 195 to keep it more simple?

File: dada2_workflow.Rmd [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+taxa <- DataFrame(taxa)
```
Comment (antagomir) [inline dada2_workflow.Rmd:192]: on line 196?

File: dada2_workflow.Rmd [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+
+tse <- TreeSummarizedExperiment(assays = assays,
+                                colData = samdf,
+                                rowData = taxa
+                                )
+tse <- tse[ , colnames(tse) != "mock"]
```
Comment (antagomir) [inline dada2_workflow.Rmd:198]: add comment what this does

File: 04_containers.Rmd [hunk 9]
```diff
@@ -233,6 +233,94 @@ Table: (\#tab:options) **Recommended options for storing multiple data tables in
 
 ## Loading experimental microbiome data
 
+### 16S workflow
+
+Result of shotgun sequencing is large number of files that include all the sequences
```
[the remark this answers is not in the document]
Comment (antagomir) [reply 04_containers.Rmd:238]: Yep.

[the remark this answers is not in the document]
Comment (TuomasBorman) [reply 04_containers.Rmd]: Thanks, I fixed that

**Pull request merged**